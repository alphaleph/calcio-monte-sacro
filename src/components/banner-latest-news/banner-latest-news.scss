@import '../../pages/globals.scss';

.banner-latest-news {
    border-top: 1px solid $holy-yellow-team;
    border-bottom: 1px solid $holy-yellow-team;
}

.banner-latest-news-list {
    margin: 2rem 0rem;
}

.banner-latest-news-button:hover,
.banner-latest-news-button:focus {
    color: $holy-yellow-team;
    text-decoration: underline 2px;
}