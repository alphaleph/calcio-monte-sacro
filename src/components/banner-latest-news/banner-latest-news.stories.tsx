import React from 'react';
import { storiesOf } from '@storybook/react';
import { BannerLatestNews } from './banner-latest-news';
import '../../pages/index.scss';

storiesOf('BannerLatestNews', module).add('default', () => (
    <BannerLatestNews />
));
