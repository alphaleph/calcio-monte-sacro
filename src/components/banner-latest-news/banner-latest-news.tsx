import React from 'react';
import { Link } from 'gatsby';
import { NewsList } from '../news-list/news-list';
import { PostsList } from '../posts-list/posts-list';
import {
    mockNewsDataNodes,
    mockFluidImageDataNodes,
} from '../__tests__/test-util';
import { SITE_URL } from '../../constants';
import './banner-latest-news.scss';

export const BannerLatestNews = () => {
    return (
        <section className="banner-latest-news section has-background-beige-main is-bold">
            <div className="container">
                <h2 className="title is-family-secondary has-text-roma-red is-2">
                    LATEST NEWS
                </h2>
                <div className="banner-latest-news-list">
                    <NewsList size={3} />
                </div>
                <div className="columns">
                    <div className="column is-half is-offset-one-quarter">
                        <Link
                            to={`/news`}
                            className="banner-latest-news-button"
                        >
                            <button
                                className="button is-fullwidth has-background-roma-red has-text-white-main is-size-5"
                                type="button"
                            >
                                <strong>SEE MORE NEWS</strong>
                            </button>
                        </Link>
                    </div>
                </div>
            </div>
        </section>
    );
};

export const TestBannerLatestNews = () => {
    return (
        <section className="banner-latest-news section has-background-beige-main is-bold">
            <div className="container">
                <h2 className="title is-family-secondary has-text-roma-red is-2">
                    LATEST NEWS
                </h2>
                <div className="banner-latest-news-list">
                    <PostsList
                        size={3}
                        posts={mockNewsDataNodes}
                        images={mockFluidImageDataNodes}
                    />
                </div>
                <div className="columns">
                    <div className="column is-half is-offset-one-quarter">
                        <Link
                            to={`${SITE_URL}/news`}
                            className="banner-latest-news-button"
                        >
                            <button
                                className="button is-fullwidth has-background-roma-red has-text-white-main is-size-5"
                                type="button"
                            >
                                <strong>SEE MORE NEWS</strong>
                            </button>
                        </Link>
                    </div>
                </div>
            </div>
        </section>
    );
};
