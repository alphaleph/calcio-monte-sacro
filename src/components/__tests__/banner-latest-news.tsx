import React from 'react';
import { BannerLatestNews } from '../banner-latest-news/banner-latest-news';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

describe('Banner Latest News', () => {
    it('renders without crashing', () => {
        const { getByText } = render(<BannerLatestNews />);
        expect(getByText('LATEST NEWS')).toBeInTheDocument();
        expect(getByText('SEE MORE NEWS')).toBeInTheDocument();
    });

    it('renders correctly', () => {
        const { getByText } = render(<BannerLatestNews />);
        expect(getByText('LATEST NEWS')).toBeVisible();
        expect(getByText('SEE MORE NEWS')).toBeVisible();
    });

    it('has not changed', () => {
        const { asFragment } = render(<BannerLatestNews />);
        expect(asFragment()).toMatchSnapshot();
    });
});
