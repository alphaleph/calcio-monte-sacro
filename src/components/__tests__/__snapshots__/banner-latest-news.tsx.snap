// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner Latest News has not changed 1`] = `
<DocumentFragment>
  <section
    class="banner-latest-news section has-background-beige-main is-bold"
  >
    <div
      class="container"
    >
      <h2
        class="title is-family-secondary has-text-black-main is-2"
      >
        LATEST NEWS
      </h2>
      <div
        class="banner-news-list"
      >
        <p>
          NewsList Here
        </p>
      </div>
      <div
        class="columns"
      >
        <div
          class="column is-half is-offset-one-quarter"
        >
          <a
            href="https://www.calciomontesacro.com/news"
          >
            <button
              class="button is-fullwidth has-background-roma-red has-text-white-main is-size-5"
              type="button"
            >
              <strong>
                SEE MORE NEWS
              </strong>
            </button>
          </a>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
